app/routes.py

```diff
@@ -1,2 +1,275 @@
+
+    elif request.method == "PUT":
+        form_data = request.get_json()
+
+        task.title = form_data["title"]
+        task.description = form_data["description"]
+        task.completed_at = form_data["completed_at"]
+        db.session.commit()
+
+        return jsonify({
+            "task": {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": task.task_completed()
+            }
+        }), 200
+
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+        return jsonify({"details": f'Task {task.task_id} "{task.title}" successfully deleted'}), 200 
+
+def slack_message_bot(task):
+
+    params = {
+        "channel": "task-notifications",
+        "text": f"Someone just completed the task {task.task_id}"
+    }
+
+    headers = {
+        "Authorization": f"Bearer {os.environ.get('SLACK_API_KEY')}"
+    }
+
+    requests.post("https://slack.com/api/chat.postMessage", data=params, headers=headers)
+
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete(task_id):
+    task = Task.query.get(task_id)
+    
+    if task == None:
+        return make_response(), 404
+    
+    task.completed_at = datetime.utcnow()
+    db.session.commit()
+
+    slack_message_bot(task)
+
+    return jsonify({
+        "task": {
+        "id": task.task_id,
+        "title": task.title,
+        "description": task.description,
+        "is_complete": task.task_completed()
+        }
+        }), 200
+    
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete(task_id):
+    task = Task.query.get(task_id)
+
+    if task == None:
+        return make_response(), 404
+    
+    task.completed_at = None
+    db.session.commit()
+
+    return jsonify({
+            "task": {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": task.task_completed()
+            }
+            }), 200
+
+goals_bp = Blueprint("goals", __name__, url_prefix= "/goals")
+@goals_bp.route("", methods=["POST", "GET"])
+def create_goal():
+    if request.method == "POST":
+        request_body = request.get_json()
+        if "title" not in request_body:
+            return jsonify({"details": "Invalid data"}), 400
+        else:
+            new_goal = Goal(title=request_body["title"])
+
+            db.session.add(new_goal)
+            db.session.commit()
+
+        return make_response({
+            "goal": {
+            "id": new_goal.goal_id,
+            "title": new_goal.title,
+            }
+        }, 201)
+    
+    elif request.method == "GET":
+        goals = Goal.query.all()
+        goals_response = []
+
+        for goal in goals:
+            goals_response.append({
+                "id": goal.goal_id,
+                "title": goal.title
+            })
```

Comment:
Just as with the task response body, note that the goal response body is used a few times. Consider moving this logic to an instance method on the goal model that returns the needed dictionary, and then calling it where needed with something like ```goal.to_json()```